### PR TITLE
doc: Add guide to releasing new versions

### PR DESCRIPTION
**Pull request by krig**

### CODE REVIEW

File: doc/releasing-a-new-version.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+1. Updating the changelog
+
+In `/ChangeLog`, there is a curated list of changes included in this
+release. This log should be updated based on the git history. Remove
+any updates that are tagged `dev:` or `test:` since these are internal
+changes, and clean up the changelog in any other way you might want.
+
+2. Tagging the release
+
+Using `git tag list` you can see the list of existing tags. Depending
```
Comment (angelabriel) [inline doc/releasing-a-new-version.md]: 'git tag' gives the list of tags. 'git tag list' creates a 'tag' named 'list'

File: doc/releasing-a-new-version.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+The patch version number indicates bugfixes only, and no breaking
+changes.
+
+## Steps
+
+1. Updating the changelog
+
+In `/ChangeLog`, there is a curated list of changes included in this
+release. This log should be updated based on the git history. Remove
```
Comment (angelabriel) [inline doc/releasing-a-new-version.md]: Is there a handy 'git log' command to get the information for the changelog? Or is it a simple 'git log' and then pick and put to the changelog file?

File: doc/releasing-a-new-version.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    git tag -s -u <user@example.com> 4.1.0
+    
+In the tag message I usually just put
+
+    Release 4.1.0
+    
+(of course, change `4.1.0` to whatever release it is you are tagging.
+
+NOTE: Remember to push the tag to the Github repository.
```
Comment (angelabriel) [inline doc/releasing-a-new-version.md]: Should we add the needed command here? So we get a more or less 'cookbook' feeling.

File: doc/releasing-a-new-version.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+The source code can be downloaded from Github:
+
+* https://github.com/ClusterLabs/crmsh/releases/tag/<VERSION>
+
+Packaged versions of crmsh should be available shortly from your
+distribution of choice. Development packages for openSUSE Tumbleweed
+are available from the Open Build System, here:
+
+* http://download.opensuse.org/repositories/network:/ha-clustering:/Factory/
```
Comment (angelabriel) [inline doc/releasing-a-new-version.md]: Is this build triggered automatically or do we need to trigger the build manually before sending the release email?

File: doc/releasing-a-new-version.md (hunk 1 above)
Comment (krig) [reply doc/releasing-a-new-version.md]: ah yes, oops :D

File: doc/releasing-a-new-version.md (hunk 2 above)
Comment (krig) [reply doc/releasing-a-new-version.md]: I have some tricks that I do, I'll put them in the guide. But basically I use `git log` with a custom format string.

Comment (krig) [reply doc/releasing-a-new-version.md]: ...and then filter manually.

File: doc/releasing-a-new-version.md (hunk 4 above)
Comment (krig) [reply doc/releasing-a-new-version.md]: This is manual at the moment. I'll add it to the guide.

File: doc/releasing-a-new-version.md (hunk 3 above)
Comment (krig) [reply doc/releasing-a-new-version.md]: Done!

**Pull request merged**